crates/contracts/solidity/Trader.sol

```diff
@@ -9,7 +9,11 @@ import { SafeERC20 } from "./libraries/SafeERC20.sol";
 import { Spardose } from "./Spardose.sol";
 
 /// @title A contract for impersonating a trader.
-contract Trader {
+/// Because this contract code gets put at the address of a trader account it uses
+/// a custom storage layout to avoid storage slot conflicts with trader accounts
+/// that are smart contracts using the default layout.
+/// layout at uint256(keccak256("cowprotocol/services trader impersonator")) - 1
+contract Trader layout at 0x02565dba7d68dcbed629110024b7b5e785bfc1a484602045eea513de8a2dcf98 {
```

Comment:
Now that you use the storage layout feature - you can get rid of `ALREADY_CALLED_SLOT` and implement it as:

```
bool private _alreadyCalled;

function alreadyCalled() private returns (bool value) {
    value = _alreadyCalled;
    _alreadyCalled = true;
}
```